### PR TITLE
[Breaking Change] Add default value to Privacy Policy Link and Terms of Service Link

### PR DESCRIPTION
**Pull request by andychow326**
ref #2500

### CODE REVIEW

File: resources/authgear/templates/en/web/__toc_pp_footer.html [hunk 1]
```diff
@@ -1,7 +1,26 @@
 {{ define "__toc_pp_footer.html" -}}
-{{ if ($.Translations.HasKey "toc-pp-footer") }}
+{{ $showFooter := $.Translations.HasKey "toc-pp-footer" }}
+{{ $termsOnly := $.Translations.HasKey "terms-of-service-link" }}
+{{ $privacyOnly := $.Translations.HasKey "privacy-policy-link" }}
+
+{{ $varient := "" }}
```
Comment (louischan-oursky) [inline resources/authgear/templates/en/web/__toc_pp_footer.html:6]: s/varient/variant

File: migrations/portal/20221003171433-migrate_toc_pp_footer.sql [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+				WHERE file_name ~ 'translation.json'
+			) AS phase1
+		) AS phase2
+	) AS phase3
+) AS phase4
+WHERE phase4.id = t1.id;
+
+-- +migrate Down
+
+-- No thing needed to be migrated down.
```
Comment (louischan-oursky) [inline migrations/portal/20221003171433-migrate_toc_pp_footer.sql:90]: Write a migration command, not a migration SQL.

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

File: resources/authgear/templates/en/web/__toc_pp_footer.html (hunk 1 above)
Comment (andychow326) [reply resources/authgear/templates/en/web/__toc_pp_footer.html]: @louischan-oursky updated :adore2x:

File: migrations/portal/20221003171433-migrate_toc_pp_footer.sql (hunk 2 above)
Comment (andychow326) [reply migrations/portal/20221003171433-migrate_toc_pp_footer.sql]: @louischan-oursky updated :adore2x:

File: cmd/portal/cmd/cmdinternal/migrate_resources_migrate_toc_pp_footer.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+	m := make(map[string]interface{})
+	err = yaml.Unmarshal(decodedConfig, &m)
+	if err != nil {
+		return fmt.Errorf("failed unmarshal yaml: %w", err)
+	}
+
+	var supportedLanguageTags []string
+
+	// Get all supported languages tags from the config
+	emptyJson, err := json.Marshal(map[string]interface{}{})
```
Comment (louischan-oursky) [inline cmd/portal/cmd/cmdinternal/migrate_resources_migrate_toc_pp_footer.go:66]: unchecked err.

File: resources/authgear/templates/en/web/__toc_pp_footer.html [hunk 4]
```diff
@@ -1,7 +1,26 @@
...
+{{ if and ($termsOnly) ($privacyOnly) }}
+  {{ $variant = "both" }}
+{{ end }}
+{{ if and (not $termsOnly) (not $privacyOnly) }}
+  {{ $showFooter = false }}
+{{ end }}
+
 <p class="text-sm primary-txt m-0">
-{{ template "toc-pp-footer" (dict "termsOfService" (.Translations.RenderText "terms-of-service-link" nil) "privacyPolicy" (.Translations.RenderText "privacy-policy-link" nil)) }}
+  {{ template "toc-pp-footer" (dict "showFooter" $showFooter
```
Comment (louischan-oursky) [inline resources/authgear/templates/en/web/__toc_pp_footer.html:21]: showFooter should be used in a if statement. In this current usage, we will render an empty p.

File: cmd/portal/cmd/cmdinternal/migrate_resources_migrate_toc_pp_footer.go (hunk 3 above)
Comment (andychow326) [reply cmd/portal/cmd/cmdinternal/migrate_resources_migrate_toc_pp_footer.go]: @louischan-oursky updated :adore2x:

File: resources/authgear/templates/en/web/__toc_pp_footer.html (hunk 4 above)
Comment (andychow326) [reply resources/authgear/templates/en/web/__toc_pp_footer.html]: @louischan-oursky updated :adore2x:

**Pull request merged**